### PR TITLE
chore: cherry-pick FP8

### PR DESCRIPTION
**Pull request by peri044**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: dev_dep_versions.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
 __version__: "2.4.0.dev0"
 __cuda_version__: "12.1"
 __tensorrt_version__: "10.0.1"
+__torch_version__: "2.3.0"
```
Comment (narendasan) [inline dev_dep_versions.yml:4]: Wrong version

Comment (narendasan) [reply dev_dep_versions.yml:4]: Also not sure if this should be in dev deps since main is always on latest nightly

File: dev_dep_versions.yml [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+# torchvision version here is not a direct dependency but the one used during testing
+__torchvision_version__: "0.18.0"
+__index_url__: "https://download.pytorch.org/whl/test/"
```
Comment (narendasan) [inline dev_dep_versions.yml:7]: Why does this need to be in the yml file?

File: packaging/pre_build_script.sh [hunk 3]
```diff
@@ -3,7 +3,6 @@
 # Install dependencies
 python3 -m pip install pyyaml
 yum install -y ninja-build gettext
-TRT_VERSION=$(python3 -c "import versions; versions.tensorrt_version()")
```
Comment (narendasan) [inline packaging/pre_build_script.sh:6]: Why was this deleted?

File: py/torch_tensorrt/_enums.py [hunk 4]
```diff
@@ -24,9 +23,9 @@ class dtype(Enum):
     f32 = auto()
     f64 = auto()
     b = auto()
+
+    f8 = auto()
```
Comment (narendasan) [inline py/torch_tensorrt/_enums.py:27]: Needs to be after bf16 to maintain the same enum values

File: py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py [hunk 5]
```diff
@@ -106,8 +106,6 @@ def __init__(
             [dtype._from(o) for o in output_dtypes] if output_dtypes else None
         )
 
-        _LOGGER.debug(f"Graph to be compiled to TensorRT: {self.module.graph}")
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py:109]: Why is this being deleted

File: py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py [hunk 6]
```diff
@@ -243,6 +241,8 @@ def _populate_trt_builder_config(
         if dtype.int8 in self.compilation_settings.enabled_precisions:
             builder_config.set_flag(trt.BuilderFlag.INT8)
 
+        if dtype.fp8 in self.compilation_settings.enabled_precisions:
+            builder_config.set_flag(trt.BuilderFlag.FP8)
         if dtype.bfloat16 in self.compilation_settings.enabled_precisions:
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py:246]: Style thing add a new line here

File: packaging/pre_build_script.sh (hunk 3 above)
Comment (peri044) [reply packaging/pre_build_script.sh:6]: This is unused variable

File: dev_dep_versions.yml [hunk 7]
```diff
@@ -1,3 +1,7 @@
 __version__: "2.4.0.dev0"
 __cuda_version__: "12.1"
 __tensorrt_version__: "10.0.1"
+__torch_version__: "2.3.0"
+# torchvision version here is not a direct dependency but the one used during testing
+__torchvision_version__: "0.18.0"
```
[the remark this answers is not in the document]
Comment (peri044) [reply dev_dep_versions.yml:6]: removed it. These are redundant changes.

File: dev_dep_versions.yml (hunk 1 above)
Comment (peri044) [reply dev_dep_versions.yml:4]: removed it. These are redundant changes.
